lib/AppleNote.rb

```diff
@@ -383,20 +384,31 @@ def all_embedded_objects
     @embedded_objects + @embedded_objects_recursive
   end
 
+  ##
+  # Unique ID for the note — prefer UUID if available, fall back to database ID
+  def unique_id
+    return @unique_id if defined?(@unique_id)
+    @unique_id = uuid.empty? ? note_id : uuid
+  end
+
+  ##
+  # Generate a file name for exporting this note to an HTML file
+  def title_as_filename(ext = '')
+    "#{unique_id} - #{title.tr('/:', '_')}#{ext}"
+  end
+
   ##
   # This method generates HTML to represent this Note, its 
   # metadata, and its contents, if applicable. It does not generate 
   # full HTML, just enough for this note's card to be displayed.
-  def generate_html
-
-    # Bail quickly if we've ever taken the time to build this before
-    return @html if @html
+  def generate_html(individual_files = false)
+    folder_href = individual_files ? "index.html" : "#folder_#{@folder.primary_key}"
 
     builder = Nokogiri::HTML::Builder.new(encoding: "utf-8") do |doc|
       doc.div {
         doc.h1 {
-          doc.a(id: "note_#{@note_id}") {
-            doc.text "Note #{@note_id}#{" (📌)" if @is_pinned}"
+          doc.a(id: "note_#{unique_id}") {
+            doc.text "Note #{unique_id}#{" (📌)" if @is_pinned}"
```

Comment:
I like this improvement, even if unique_id ends up being a CLI switch. Much cleaner, thanks!